image_processing/src/app.py

```diff
@@ -0,0 +1,100 @@
+from h2o_wave import Q, app, ui, main
+import ip_utils as ip
+import ui_utils as U
+import helper
+import cv2
+import os
+import pandas as pd
+import views.layout_utils as layouts
+
+
+@app('/ip')
+async def serve(q: Q):
+    _hash = q.args['#']
+
+    if not q.client.flag:
+        q.client.image = cv2.imread('../data/lena.jpg')[:, :, ::-1]
+        q.client.image_df = pd.DataFrame({
+            'Image': [],
+            'Timestamp': []
+        })
+        q.client.columns = [
+            ui.table_column(name='images_col', label='Image', searchable=True, sortable=True, link=True),
+            ui.table_column(name='timestamp_col', label='Timestamp', searchable=True, sortable=True),
+        ]
+
+        q.client.flag = True
+
+    if _hash:
+        if _hash == 'menu/translation':
+            await layouts.translation_layout(q)
+        elif _hash == 'menu/rotation':
+            await layouts.rotation_layout(q)
+        elif _hash == 'menu/rgb2gray':
+            await layouts.rgb2gray_layout(q)
+        elif _hash == 'menu/histogram':
+            await layouts.histogram_layout(q)
+        elif _hash == 'menu/blur':
+            await layouts.blur_layout(q)
+        elif _hash == 'menu/histogram_match':
+            await layouts.histogram_match_layout(q)
+            q.client.load_image_hist_eq = True
+            q.client.load_count = 0
+        elif _hash == 'menu/edge_detection':
+            await layouts.edge_detection(q)
+
+    if q.args.image_table:
+        await U.load_image(q)
+
+    if q.args.image_upload:
+        q.client.image_df = helper.update_df(q.client.image_df, q.args.image_upload[-1])
+        q.page['main_sidebar'].items[2].table.rows = [ui.table_row(
+            name=row.Image,
+            cells=[row.Image, row.Timestamp]
+        ) for row in q.client.image_df.itertuples()]
+
+    if q.args.translation:
```

Comment:
It might be nice to use `on` and `handle_on` since you have many checks which immediately call a function: https://h2oai.github.io/wave/docs/examples/routing/